applications/GeoMechanicsApplication/tests/cpp_tests/custom_processes/test_fix_water_pressures_above_phreatic_line.cpp

```diff
@@ -0,0 +1,244 @@
+                                               "of the phreatic line must be non-empty vectors.";
+    KRATOS_EXPECT_EXCEPTION_IS_THROWN(
+        FixWaterPressuresAbovePhreaticLineProcess(r_model_part, test_parameters), expected_error_message);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(TestFixWaterPressureAbovePhreaticLine_ThrowsUponConstructionWhenXYCoordinatesAreNotEqualLength,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    constexpr double y_node_1     = 0.0;
+    constexpr double y_node_2     = -2.0;
+    auto             model        = Model{};
+    auto&            r_model_part = CreateModelPartWithTwoNodesAtHeights(model, y_node_1, y_node_2);
+
+    const auto test_parameters = Parameters{R"(
+            {
+                "model_part_name": "foo",
+                "x_coordinates": [1.0],
+                "y_coordinates": [2.0, 3.0]
+            }  )"};
+
+    // Act & Assert
+    const std::string expected_error_message = "The lengths of the x_coordinates and y_coordinates "
+                                               "of the phreatic line must be equal.";
+    KRATOS_EXPECT_EXCEPTION_IS_THROWN(
+        FixWaterPressuresAbovePhreaticLineProcess(r_model_part, test_parameters), expected_error_message);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(TestFixWaterPressureAbovePhreaticLine_FixesAllWaterPressuresToZeroWhenAllNodesAbovePhreaticLine,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    constexpr double y_node_1     = 0.0;
+    constexpr double y_node_2     = -1.0;
+    auto             model        = Model{};
+    auto&            r_model_part = CreateModelPartWithTwoNodesAtHeights(model, y_node_1, y_node_2);
+    auto             test_parameters = CreateParametersWithConstantPhreaticLineAtHeight(-2.0);
+    r_model_part.GetNode(1).FastGetSolutionStepValue(WATER_PRESSURE) = 1.0;
+    r_model_part.GetNode(2).FastGetSolutionStepValue(WATER_PRESSURE) = 2.0;
+    FixWaterPressuresAbovePhreaticLineProcess process(r_model_part, test_parameters);
+
+    // Act
+    process.ExecuteInitializeSolutionStep();
+
+    // Assert
+    for (auto& r_node : r_model_part.Nodes()) {
+        EXPECT_DOUBLE_EQ(r_node.FastGetSolutionStepValue(WATER_PRESSURE), 0.0);
+        EXPECT_TRUE(r_node.IsFixed(WATER_PRESSURE));
+    }
+}
+
+KRATOS_TEST_CASE_IN_SUITE(TestFixWaterPressureAbovePhreaticLine_DoesNothingWhenAllNodesBelowPhreaticLine,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    constexpr double y_node_1     = 0.0;
+    constexpr double y_node_2     = -1.0;
+    auto             model        = Model{};
+    auto&            r_model_part = CreateModelPartWithTwoNodesAtHeights(model, y_node_1, y_node_2);
+    r_model_part.GetNode(1).FastGetSolutionStepValue(WATER_PRESSURE) = 1.0;
+    r_model_part.GetNode(2).FastGetSolutionStepValue(WATER_PRESSURE) = 2.0;
+
+    auto test_parameters = CreateParametersWithConstantPhreaticLineAtHeight(1.0);
```

Comment:
Same question here: can we make this variable `const`?
```suggestion
    const auto test_parameters = CreateParametersWithConstantPhreaticLineAtHeight(1.0);
```

Reply:
Done